### PR TITLE
Feat(TextArea AutoResize) - Adding autoResize function to textArea

### PR DESCRIPTION
**Pull request by matheusrpp**
## Description
Put the description of the task here.

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] TypeScript updated

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

### CODE REVIEW

File: components/TextArea/TextArea.jsx [hunk 1]
```diff
@@ -31,22 +37,50 @@ class TextArea extends Component {
...
     this._id = id || ID_GENERATOR.next().value;
+    this._fontSize = 16;
+    this._linHeight = 1.5;
   }
 
   onChangeTextArea = ev => {
     const { onChange } = this.props;
+    const { minRows, maxRows } = this.state;
+    const textareaLineHeightInPx = this._fontSize * this._linHeight;
+    const previousRows = ev.target.rows;
```
Comment (rapahaeru) [inline components/TextArea/TextArea.jsx:63]: how about of using destructuting here, to make code cleaner?
here my suggestions:
- i updated the contant names
- renamed the event arg
- and i applied a desctructuring in ev.target.
```
  onChangeTextArea = event => {
    const { target } = event;
    const { onChange } = this.props;
    const { minRows, maxRows } = this.state;
    const textareaLineHeightInPx = this._fontSize * this._lineHeight;
    const previousNumberRows = target.rows;
    target.rows = minRows;

    const currentNumberRows = Math.floor(
      target.scrollHeight / textareaLineHeightInPx,
    );
    
    if (currentNumberRows === previousNumberRows) {
      target.rows = currentNumberRows;
    }

    if (currentNumberRows >= maxRows) {
      target.rows = maxRows;
      target.scrollTop = target.scrollHeight;
    }

    const inputValue = event.currentTarget.value;
    this.setState({
      currentValue: inputValue,
      hasDefaultValue: null,
      rows: currentNumberRows < maxRows ? currentNumberRows : maxRows,
    });

    onChange(event);
  };
```

File: components/TextArea/TextArea.jsx [hunk 2]
```diff
@@ -97,6 +135,13 @@ class TextArea extends Component {
 }
 
 TextArea.defaultProps = {
+  /** Disables the default resize and activates the auto resize */
+  autoResize: false,
+  autoResizeConfig: {
```
Comment (rapahaeru) [inline components/TextArea/TextArea.jsx:140]: how about of:
```
  isAutoResize: false,
  autoResizeConfig: {
    initialRows: 1,
    minRows: 1,
    maxRows: 5,
  },
```

File: stories/TextArea/TextArea.regression-test.story.jsx [hunk 3]
```diff
@@ -87,3 +87,8 @@ PlaceholderWithDarkSkin.parameters = parameters;
 PlaceholderWithDarkSkin.args = {
   placeholder: 'placeholder',
 };
+
+export const AutoResize = Template.bind({});
+AutoResize.args = {
+  autoResize: true,
+};
```
Comment (rapahaeru) [inline stories/TextArea/TextArea.regression-test.story.jsx:94]: how about of a second example using the props?
```
export const AutoResizeWithMax3 = Template.bind({});
AutoResizeWithMax3.args = {
  autoResize: true,
  autoResizeConfig: {
    rowsConfig: 1,
    minRowsConfig: 1,
    maxRowsConfig: 3,
  },
};
```

**Review by rapahaeru — CHANGES_REQUESTED** [review]
Congrats!
the new dynamic it was good.

## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [x] TypeScript updated

### Browsers review
- [x] Layout review
- [ ] Ux review validation

File: components/TextArea/index.d.ts [hunk 4]
```diff
@@ -2,6 +2,12 @@ import { Component, ChangeEventHandler, TextareaHTMLAttributes } from 'react';
 
 export interface TextAreaProps extends TextareaHTMLAttributes<HTMLTextAreaElement> {
     disabled?: boolean;
+    autoResize?: boolean;
```
Comment (rapahaeru) [inline components/TextArea/index.d.ts:5]: i think you missed the update properties here

**Review by rapahaeru — CHANGES_REQUESTED** [review]

File: components/TextArea/TextArea.jsx [hunk 5]
```diff
@@ -111,6 +161,12 @@ TextArea.defaultProps = {
 };
 
 TextArea.propTypes = {
+  isAutoResize: PropTypes.bool,
+  autoResizeConfig: PropTypes.shape({
+    initialRows: PropTypes.number,
+    minRows: PropTypes.number,
+    maxRows: PropTypes.number,
+  }),
```
Comment (toncabral88) [inline components/TextArea/TextArea.jsx:169]: How about check if `autoResizeConfig` is different of undefined to check if component is auto resize?

**Review by toncabral88 — COMMENTED** [review]
Very nice man!

- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review
- [x] TypeScript updated

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**